### DISCUSSION TITLE
Partial Results

### DISCUSSION THREAD

**Discussion by andresmonc** [Netflix/dgs-framework#1221] [Q&A]
Hello - I'm trying to figure out how I could return partial results as well as errors.

Imagine the method below is exposed to update greetings
```
	@PreAuthorize("hasAuthority('SCOPE_update:greetings')")
	@DgsMutation
	public List<Greeting> updateGreetings(List<BigInteger> ids) {
		return greetingService.updateGreetings(ids);
	}
```

now in the underlying service method we loop through the id's in the list and attempt to update each one individually
```
    public List<Greeting> updateInMemoryGreeting(List<BigInteger> ids ) {
        List<Greeting> greetings = new ArrayList<>()
        ids.forEach(bigInteger -> {
            try {
                int updatedRowCount = greetingRepository.setGreetingMessageById(...);
                if (updatedRowCount == 0) {
                    throw new GreetingNotUpdatedException("greeting not updated");
                }
                return new Greeting(..);
            } catch (Exception e) {
                throw new GreetingNotUpdatedException(e.getMessage());
            }
        });
        return greetings;
    }
```

Currently if I throw an exception any partial results are thrown away and only the exception is returned. Is there any way to return both partial results and errors?

**Comment by srinivasankavitha** [comment]
You can return a DataFetcherResult from your datafetcher with the custom error as described here: https://www.graphql-java.com/documentation/execution#returning-data-and-errors
